extractor/extractor.py

```diff
@@ -0,0 +1,205 @@
+        self.soup = ''
+        self.contents_list = contents.Contents()
+
+    def parse(self):
+        global req_session
+
+        req_session = requests.Session()
+        file = req_session.get(self.url).text    # To request Html from URL
+
+        # Remove the comment to read local files for testing purpose
+        # file = open('./test_html.html','r').read()
+
+        self.soup = BeautifulSoup(file,'lxml')
+        self.clean_soup()
+
+        self.extract()
+
+    def clean_soup(self):
+        decomposable_tags = self.soup.find_all(['script', 'noscript', 'style', 'header', 'footer', 'meta', 'link'])
+        comments = self.soup.find_all(text=lambda text: isinstance(text, Comment))
+        ads = self.soup.find_all(re.compile('.*'), {'class': re.compile('(^ad-|^ads-)')})
+
+        comments = [comment.extract() for comment in comments]
+        [tag.decompose() for tag in decomposable_tags]
+        [ad.decompose() for ad in ads]
+
+    def extract(self):
+        # Extract data from head
+        head = self.soup.find('head')
+        self.extract_title_tag(head)
+
+        # Extract data from body
+        body_soup = self.soup.find('body')
+        self.iterate_over_dom(body_soup)
+
+    def extract_title_tag(self, soup_head):
+        title = soup_head.title
+        _content = contents._Content()
+        _content.contentType = 'Text'
+        _content.text.type = 'title'
+        _content.text.text_string = title.get_text()
+        self.contents_list.add_content(_content)
+
+    def iterate_over_dom(self, node):
+        isContent, _content = self.is_content(node)
+
+        # Add to the final list if a valid content is extracted
+        if isContent:
+            self.contents_list.add_content(_content)
+
+        if node is not None and node.name is not None and not isContent:
+            for child in node.contents:
+                self.iterate_over_dom(child)
+
+    def is_content(self, node):
+        _content = {}
+
+        # check if content is text
+        isText, text = self.is_text(node)
+        if isText:
+            _content = contents._Content()
+            _content.contentType = 'Text'
+            _content.text = text
+            return True, _content
+
+        # check if content is image
+        isImage, image = self.is_image(node)
+        if isImage:
+            _content = contents._Content()
+            _content.contentType = 'Image'
+            _content.image = image
+            return True, _content
+
+        # check if content is video
+        isVideo, video = self.is_video(node)
+        if isVideo:
+            _content = contents._Content()
+            _content.contentType = 'Video'
+            _content.video = video
+            return True, _content
+
+        return False, {}
+
+    def empty_text(self, _str):
+        return _str == '\n' or _str == ' ' or len(_str) == 0
+
+    def not_important_text(self, node):
+        heading_tags = {'h1', 'h2', 'h3', 'h4', 'h5', 'h6'}
```

Comment:
Make `heading_tags` global instead of creating on every function call. Same with `all_text_tags` down below.